source/slang/slang-compiler.cpp

```diff
@@ -2177,6 +2177,103 @@ namespace Slang
         return SLANG_OK;
     }
 
+    static void writeString(Stream& stream, const char* string)
+    {
+        stream.write(string, strlen(string));
+    }
+
+    static void escapeDependencyString(const char* string, StringBuilder& builder)
+    {
+        // make has unusual escaping rules, but we only care about characters that are acceptable in a path
+        for (const char* p = string; *p; ++p)
+        {
+            char c = *p;
+            switch(c)
+            {
+            case ' ':
+            case ':':
+            case '#':
+            case '[':
+            case ']':
+            case '\\':
+                builder.appendChar('\\');
+                break;
+
+            case '$':
+                builder.appendChar('$');
+                break;
+            }
+
+            builder.appendChar(c);
+        }
+    }
+
+    // Writes a line to the file stream, formatted like this:
+    //   <output-file>: <dependency-file> <dependency-file...>
+    static void writeDependencyStatement(Stream& stream, EndToEndCompileRequest* compileRequest, const String& outputPath)
```

Comment:
Nitpik: By convention (although certainly not always used by any means) for functions that are local to a .cpp file (or are not part of the public interface of a class) we prefix with _. So `_writeDependencyStatement` here.

Reply:
Done.